### PR TITLE
Match most of cpu.c (+ merge and match tree.c)

### PR DESCRIPTION
**Pull request by Yanis42**
this got some non-matchings but overall it's matched, also I merged the tree functions there like it is on oot-gc

### CODE REVIEW

File: src/emulator/cpu.c [hunk 1]
```diff
@@ -1,7580 +1,7743 @@
...
- * 
- * @param cpu The emulated VR4300.
- * @param new_dev_idx A pointer to the index in the cpu->devices array which the device was created.
- * @param dev_obj The object which will handle reuqests for this device.
- * @param vaddr Starting address of the device's address space.
- * @param paddr Starting physical address of the device's address space.
- * @param size Size of the device's memory space.
- * @param create_arg An argument which will be passed back to the device's event handler.
- * @return s32 1 on success, 0 otherwise.
- */
```
Comment (cadmic) [inline src/emulator/cpu.c:337]: Should we preserve these comments? Seems kinda sad to lose them, although the param names would need to be updated (and bool return values too)

File: src/emulator/cpu.c
Comment (cadmic) [inline src/emulator/cpu.c:1]: this doesn't build for me (undefined identifier 'NM8E')

Comment (Yanis42) [reply src/emulator/cpu.c:1]: should work now

File: src/emulator/cpu.c (hunk 1 above)
Comment (Yanis42) [reply src/emulator/cpu.c]: done, doing this made me found a tiny error this file had, some functions were returning ``bool`` when it should be ``s32`` (the ones that are used as a ``CpuExecuteFunc`` pointer)

**Review by cadmic — APPROVED** [review]

**Pull request merged**